torch/jit/supported_ops.py

```diff
@@ -58,70 +83,210 @@ def hidden(name):
-        for elem in dir(mod):
-            builtin = torch.jit._find_builtin(getattr(mod, elem))
-            if builtin is not None:
-                schemas = torch._C._jit_get_schemas_for_operator(builtin)
-                for schema in schemas:
-                    # remove _tan but not __and__
-                    if not hidden(elem):
-                        functions.append(emit_schema(name, elem, schema))
+    return "TorchScript Builtin Functions", functions
 
-    def is_tensor_method(schema):
-        if len(schema.arguments) == 0:
-            return False
-        self = schema.arguments[0]
-        if self.name != 'self':
-            return False
-        if not self.type.isSubtypeOf(torch._C.TensorType.get()):
-            return False
-        return True
 
-    methods = []
-    # discover methods
-    for elem in dir(torch.Tensor):
-        if not hidden(elem):
-            schemas = torch._C._jit_get_schemas_for_operator("aten::" + elem)
+def _get_math_builtins():
+    functions = []
+    builtins = filter(lambda fn: _is_math_fn(fn[0]), _get_builtins_helper())
+    builtins = list(builtins)
+    # Iterate over the specially added builtins
+    for fn, _builtin_name in builtins:
+        mod = inspect.getmodule(fn)
+        builtin = torch.jit._find_builtin(fn)
+        if builtin is not None:
+            schemas = torch._C._jit_get_schemas_for_operator(builtin)
             for schema in schemas:
-                if is_tensor_method(schema):
-                    methods.append(emit_schema('Tensor', elem, schema, arg_start=1))
+                functions.append(_emit_schema(mod.__name__, fn.__name__, schema))
+                pass
+
+    return "``math`` Module", functions
+
+
+def _get_global_builtins():
+    # Taken from the 'globals' map in torch/csrc/jit/script/compiler.cpp
+    supported_builtins = [
+        'print',
+        'tuple',
+        'float',
+        'int',
+        'bool',
+        'str',
+        'getattr',
+        'hasattr',
+        'isinstance',
+        'len',
+        'hex',
+        'oct',
+        'round',
+        'hash',
+        'min',
+        'max',
+        'abs',
+        'all',
+        'divmod',
+        'list',
+        'ord',
+        'chr',
+        'bin',
+        'range',
+        'zip',
+        'enumerate',
+        'sorted',
+    ]
+
+    op_renames = {
+        'bool': 'aten::Bool',
+        'int': 'aten::Int',
+        'float': 'aten::Float',
+        'abs': 'prim::abs',
+        'max': 'prim::max',
+        'min': 'prim::min',
+        'range': 'fake::does_not_exist',
+    }
+
+    schemaless_op_explanations = {
+        'print': 'Print any value',
+        'tuple': 'Lists cannot be converted to tuples with this method since their size is not statically known',
+        'getattr': 'Attribute name must be a literal string',
+        'hasattr': 'Attribute name must be a literal string',
+        'isinstance': 'Result is static',
+        'zip': 'Arguments must be lists',
```

Comment:
This isn't quite true. We can iterate over Tensors, Lists, Tuples, Dictionaries, Strings, and ModuleLists / ModuleDicts. We should probably have a note on iterables in torchscript, which you then reference here.